.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,223 @@
+#        run: |
+#          sudo rm /etc/docker/daemon.json
+#          echo '{"storage-opt": [ "dm.basesize=50G" ]}' | sudo tee /etc/docker/daemon.json
+#          sudo service docker restart
+
+      - name: Build deps
+        run: |
+          pushd .docker
+          docker pull qgis/qgis3-build-deps:${DOCKER_TAG}
+          docker build --cache-from qgis/qgis3-build-deps:${DOCKER_TAG} -t qgis/qgis3-build-deps:${DOCKER_TAG} -f qgis3-build-deps.dockerfile .
+          popd
+
+      - name: Push deps image
+        if: ${{ github.event_name != 'pull_request' }}
+        run: |
+          docker login -u="${DOCKER_USERNAME}" -p="${DOCKER_PASSWORD}"
+          docker push "qgis/qgis3-build-deps:${DOCKER_TAG}"
+
+      - name: Prepare build cache for pull request
+        uses: pat-s/always-upload-cache@v2.1.3
+        if: github.event_name == 'pull_request'
+        with:
+          path: /home/runner/QGIS/.ccache
+          key: build-ccache-${{ github.head_ref }}-${{ github.sha }}
+          # The head_ref or source branch of the pull request in a workflow run.
+          # The base_ref or target branch of the pull request in a workflow run.
+          restore-keys: |
+            build-ccache-${{ github.head_ref }}-
+            build-ccache-${{ github.base_ref }}-
+            build-ccache-master-
+
+      - name: Prepare build cache for branch/tag
+        # use a fork of actions/cache@v2 to upload cache even when the build or test failed
+        uses: pat-s/always-upload-cache@v2.1.3
+        if: github.event_name != 'pull_request'
+        with:
+          path: /home/runner/QGIS/.ccache
+          # The branch or tag ref that triggered the workflow run. For branches this in the format refs/heads/<branch_name>, and for tags it is refs/tags/<tag_name>
+          key: build-ccache-${{ github.ref }}-${{ github.sha }}
+          restore-keys: |
+            build-ccache-${{ github.ref }}-
+            build-ccache-master-
+
+      - name: Compile QGIS
+        id: compile
+        run: |
+          docker run -t --name qgis_container \
+                     -v $(pwd):/root/QGIS \
+                     -v /home/runner/QGIS/.ccache:/root/.ccache \
+                     --env-file .docker/docker-variables.env \
+                     --env PUSH_TO_CDASH=true \
+                     qgis/qgis3-build-deps:${DOCKER_TAG} \
+                     /root/QGIS/.docker/docker-qgis-build.sh
+          docker commit qgis_container qgis_image
+
+      - name: Push image
+        if: github.event_name == 'pull_request' && contains( github.event.pull_request.labels.*.name, 'Push Docker Image')
+        env:
+          GITHUB_PR_NUMBER: ${{github.event.number}}
+        run: |
+          docker login -u="${DOCKER_USERNAME}" -p="${DOCKER_PASSWORD}"
+          docker commit qgis_container qgis/qgis-pr-build:PR${GITHUB_PR_NUMBER}
+          docker push qgis/qgis-pr-build:PR${GITHUB_PR_NUMBER}
+
+      - name: run unit tests
```

Comment:
I'd capitalize 'run' so its consistent with the other section titles:
![image](https://user-images.githubusercontent.com/1728657/107110668-726e5980-687c-11eb-82ee-84701905abe7.png)